apps/emqx_dashboard/src/emqx_dashboard.appup.src

```diff
@@ -1,13 +1,7 @@
 %% -*- mode: erlang -*-
 %% Unless you know what you are doing, DO NOT edit manually!!
 {VSN,
-  [{"5.0.0",
-    [{load_module,emqx_dashboard,brutal_purge,soft_purge,[]},
-     {load_module,emqx_dashboard_api,brutal_purge,soft_purge,[]},
-     {load_module,emqx_dashboard_token,brutal_purge,soft_purge,[]}]},
-   {<<".*">>,[]}],
-  [{"5.0.0",
-    [{load_module,emqx_dashboard,brutal_purge,soft_purge,[]},
-     {load_module,emqx_dashboard_api,brutal_purge,soft_purge,[]},
-     {load_module,emqx_dashboard_token,brutal_purge,soft_purge,[]}]},
-   {<<".*">>,[]}]}.
+  %% we should always restart dashboard to make sure api rules/swagger is updated
+  [{<<".*">>,[{restart_application, emqx_dashboard}]}],
+  [{<<".*">>,[{restart_application, emqx_dashboard}]}]
```

Comment:
Maybe we need stop dashboard listeners.

Reply:
{apply, {emqx_dashboard, stop_listeners, []}}.

Reply:
We don't need to do this in 5.0, because we always get app env when exec upgrade code.

Why we need to `{apply, {minirest, stop_http, ['http:dashboard']}},` in 4.2.x, 
because [bin/install_upgrade.escript](https://github.com/emqx/emqx-enterprise-rel/commit/365fc93beb40e64ee71b974482586e6a955e47fe
) 's change don't include in `e4.2.0`(I guess 4.2.0 must be retag). so we can't get emqx_dashboard's app when upgrade, All app's env is ok after this commit fix.

Reply:
this also happens in 4.3.0: https://github.com/emqx/emqx/blob/4.3.0/bin/install_upgrade.escript#L250-L265
but 5.0.0 is ok now.

Reply:
Oh, it is so cool!